src/components/NavBar/NavBar.jsx

```diff
@@ -0,0 +1,169 @@
+                  </Button>
+                </Link>
+                <Box sx={{ flexGrow: 1 }} />
+                <IconButton
+                  size="large"
+                  aria-label="show 17 new notifications"
+                  color="inherit"
+                  onClick={handleNotificationsMenu}
+                >
+                  <Badge badgeContent={17} color="error">
+                    <NotificationsIcon />
+                  </Badge>
+                </IconButton>
+                <Menu
+                  id="menu-appbar-notifications"
+                  anchorEl={anchorElNotifications}
+                  anchorOrigin={{
+                    vertical: 'top',
+                    horizontal: 'right'
+                  }}
+                  keepMounted
+                  transformOrigin={{
+                    vertical: 'top',
+                    horizontal: 'right'
+                  }}
+                  open={Boolean(anchorElNotifications)}
+                  onClose={handleNotificationsClose}
+                >
+                  <MenuItem onClick={handleNotificationsClose}>
+                    <p>Notification 1</p>
+                  </MenuItem>
+                  <MenuItem onClick={handleNotificationsClose}>
+                    <p>Notification 2</p>
+                  </MenuItem>
+                </Menu>
+                <IconButton
+                  size="large"
+                  edge="end"
+                  aria-label="account of current user"
+                  aria-controls={menuId}
+                  aria-haspopup="true"
+                  onClick={handleOpenMenu}
+                  color="inherit"
+                >
+                  <AccountCircle />
+                </IconButton>
+                <Menu
+                  id="menu-appbar-notifications"
+                  anchorEl={anchorElNotifications}
+                  anchorOrigin={{
+                    vertical: 'top',
+                    horizontal: 'right'
+                  }}
+                  keepMounted
+                  transformOrigin={{
+                    vertical: 'top',
+                    horizontal: 'right'
+                  }}
+                  open={Boolean(anchorElNotifications)}
+                  onClose={handleNotificationsClose}
+                >
+                  <MenuItem onClick={handleNotificationsClose}>
+                    <p>Notification 1</p>
+                  </MenuItem>
+                  <MenuItem onClick={handleNotificationsClose}>
+                    <p>Notification 2</p>
+                  </MenuItem>
+                </Menu>
+              </>
+            </>
+          ) : (
+            <>
+              <Box sx={{ flexGrow: 1 }} />
+              <Typography variant="h4" noWrap component="div">
+                Getting started with PASS
+              </Typography>
+              <Box sx={{ flexGrow: 1 }} />
+            </>
+          )}
+        </Toolbar>
+      </AppBar>
+      {openMenu ? (
```

Comment:
**Since KaHung discussed some specifics, I'm going to look at more basics...**

The `menu for the account settings/profile` got refactored into it's own component. 
But the `menu for the notifications` did not. 

Was there a reason why one and not the other?

---

Also the `menu for the account settings/profile` seems to rely on state to render **(line 156)**, where the `notifications menu` seems to solely rely on an `onClick` event. 

Is there a reason for the difference? Did I miss something here entirely? 

---

Maybe if `Navbar` items continue to grow we could discuss need (for maintainability) to have the links be one separate component, the notifications be another, and the profile be another? Not sure that's necessary now, but if menu keeps growing/links keep growing - maybe?

That way we could just create an array of the items, in their own components, and `.map()` over them to create the lists/menus.  Less repeated code maybe.

Reply:
The above points are based on code I used from MUI templates, so my assumption is that those are at least serviceable. If we want to look into optimizing it, perhaps that can be done in a future PR?